src/TestProblem/ELBDM/DiskHeating/Init_TestProb_ELBDM_DiskHeating.cpp

```diff
@@ -0,0 +1,1208 @@
+static bool AddFixedHalo;  // add a fixed halo, must enable OPT__FREEZE_FLUID
+static bool HaloUseTable;   // 0 = from analytical profile, 1 = from table
+       double m_22;         // ELBDM particle mass, used for soliton profile of the fixed halo
+       double CoreRadius;   // soliton radius of the fixed halo (in kpc)
+static double Rho_0;        // halo rho_0 (in 1.0e+10 Msun*kpc^-3)
+static double Rs;           // halo Rs (in kpc)
+static double Alpha;        // dimensionless
+static double Beta;         // dimensionless
+static double Gamma;        // dimensionless
+static char   DensTableFile[1000];   // fixed halo density profile filename
+static int    DensNbin;              // Nbins of density table
+static double *DensTable_r = NULL;   // radius of density table
+static double *DensTable_d = NULL;   // density of density table
+static bool   AddParWhenRestart;        // add a new disk to an existing snapshot, must enable OPT__RESTART_RESET
+static bool   AddParWhenRestartByfile;  // add a new disk via PAR_IC
+static int    AddParWhenRestartNPar;    // particle number of the new disk
+static double Disk_Mass;                // total mass of the new disk
+static double Disk_R;                   // scale radius of the new disk
+static char   DispTableFile[1000];   // velocity dispersion filename
+static int    DispNbin;              // Nbins of velocity dispersion table
+static double *DispTable_r = NULL;   // radius of velocity dispersion table
+static double *DispTable_d = NULL;   // velocity dispersion of velocity dispersion table
+
+// =======================================================================================
+#ifdef PARTICLE
+void Par_Init_ByFunction_DiskHeating( const long NPar_ThisRank, const long NPar_AllRank,
+                                real *ParMass, real *ParPosX, real *ParPosY, real *ParPosZ,
+                                real *ParVelX, real *ParVelY, real *ParVelZ, real *ParTime,
+                                real *ParType, real *AllAttribute[PAR_NATT_TOTAL]);
+static void Init_NewDiskRestart();
+static void Par_AfterAcceleration( const long NPar_ThisRank, const long NPar_AllRank,
+                                 real *ParMass, real *ParPosX, real *ParPosY, real *ParPosZ,
+                                 real *ParVelX, real *ParVelY, real *ParVelZ,
+                                 real *ParAccX, real *ParAccY, real *ParAccZ,
+                                 real *ParTime, real *ParType, real *AllAttribute[PAR_NATT_TOTAL] );
+
+
+void (*Par_AfterAcceleration_Ptr)( const long NPar_ThisRank, const long NPar_AllRank,
+                                 real *ParMass, real *ParPosX, real *ParPosY, real *ParPosZ,
+                                 real *ParVelX, real *ParVelY, real *ParVelZ,
+                                 real *ParAccX, real *ParAccY, real *ParAccZ,
+                                 real *ParTime, real *ParType, real *AllAttribute[PAR_NATT_TOTAL] ) = Par_AfterAcceleration;
```

Comment:
What is the purpose of using a function pointer? Do you plan to support some variants of `Par_AfterAcceleration()`?